restli-docgen/src/main/java/com/linkedin/restli/docgen/examplegen/ExampleRequestResponseGenerator.java

```diff
@@ -1210,7 +1210,7 @@ public FixedTemplatePlaceholder(Object object, FixedDataSchema schema)
 
   private static ResourceModel buildPlaceholderResourceModel(ResourceSchema resourceSchema)
   {
-    return new ResourceModel(RecordTemplatePlaceholder.class,
```

Comment:
What's the purpose of this change?

Reply:
This is good question, so when I do gradle test, the this test class failed because in the `RecordTemplatePlaceholder` it is not a real template, so, in my newly added code, i need to get the value class.
```
routingResult.getResourceMethod().getResourceModel().getValueClass();
```
So this change is not there, and we use the `RecordTemplatePlaceholder`, we will get an exception.

Since I see the ResourceModel takes an RecordTemplate as the input, the change to use resourceSchema seems to be a perfect fit. It is needed to say, I don't know why the original code uses the RecordTemplatePlaceholder rather than the real resourceSchema

Reply:
I don't think this is a good idea. Using a random record type like `ResourceSchema` is misleading and made me believe that this logic is inherently related to resource schemas. I'd suggest you do one of two things:
1. Be explicit about using a dummy record by using something like `EmptyRecord`.
2. Fix the issue in `RecordTemplatePlaceholder`, this is an easy fix, since you can just define the schema as `new RecordDataSchema(new Name(RecordTemplatePlaceholder.class.getCanonicalName()), RecordDataSchema.RecordType.RECORD)`.

Reply:
I removed the change, it looks like I no longer need this change anymore. Will notice your suggestion next time when same situation happen